src/nvim/drawscreen.c

```diff
@@ -1849,9 +1854,9 @@ static void win_update(win_T *wp, DecorProviders *providers)
 
   // Now that the window has been redrawn with the old and new cursor line,
   // update w_last_cursorline.
-  wp->w_last_cursorline = cursorline_standout ? wp->w_cursor.lnum : 0;
+  wp->w_last_cursorline = cursorline_standout ? wp->w_cursorline : 0;
 
-  wp->w_last_cursor_lnum_rnu = wp->w_p_rnu ? wp->w_cursor.lnum : 0;
+  wp->w_last_cursor_lnum_rnu = wp->w_p_rnu ? wp->w_cursorline : 0;
```

Comment:
I don't think this change is correct. If 'cursorline' has been enabled and then disabled, `w_cursorline` will be an incorrect value.

Reply:
It fixed an issue for which I thought I added a test but it doesn't currently fail. This diff does make it fail:
```diff
+++ b/test/functional/ui/fold_spec.lua
@@ -208,7 +208,7 @@ describe("folded lines", function()
         ]])
       end
       command("set relativenumber cursorlineopt=number")
-      feed("jk")
+      feed("jgk")
       if multigrid then
         screen:expect([[
         ## grid 1
```

Reply:
It seems the updated test still fails regardless of this change, I thought I had this working before, but I guess it still requires a fix.

Reply:
Swapping around the relativenumber and cursorline if blocks in `get_line_number_attr()` fixes this case. I wonder if it makes anything else fail, I wouldn't expect it to...

Reply:
This would also fix it, any preference? I guess this is more in line with the other changes already made:
```diff
 static int get_line_number_attr(win_T *wp, linenr_T lnum, int row, int startrow, int
filler_lines)
 {
   if (wp->w_p_rnu) {
-    if (lnum < wp->w_cursor.lnum) {
+    if (lnum < wp->w_cursorline) {
       // Use LineNrAbove
       return win_hl_attr(wp, HLF_LNA);
     }
-    if (lnum > wp->w_cursor.lnum) {
+    if (lnum > wp->w_cursorline) {
       // Use LineNrBelow
       return win_hl_attr(wp, HLF_LNB);
     }
```

Reply:
There may be another bug here.

Reply:
> This would also fix it, any preference? I guess this is more in line with the other changes already made:

This one is actually incorrect when 'cursorline' is not set, reverted to the previous version.